core/domain/html_validation_service.py

```diff
@@ -415,13 +415,10 @@ def convert_to_ckeditor(html_data):
                         else:
                             break
 
-    # This block wraps p tag in li tag if the parent of p is ol/ul tag. Also,
-    # if the parent of p tag is pre tag, it unwraps the p tag.
+    # This block unwraps the p tag, if the parent of p tag is pre tag.
     for p in soup.findAll(name='p'):
         if p.parent.name == 'pre':
             p.unwrap()
-        elif p.parent.name in list_tags:
```

Comment:
@bansalnitish is this deletion legit?

@lilithxxx the usual question of explaining deletions still applies.

Reply:
I am sorry I forgot to explain the deletion here. I think this check is covered by these lines:
https://github.com/oppia/oppia/blob/8b3a89fc8197a969704bc671ff5f46df88e25cbb/core/domain/html_validation_service.py#L404

Reply:
Yeah, the deletion is legit.